backend/agent_logic/tools/sandbox_executor.py

```
@@ -1,79 +1,257 @@
+            result = container.wait(timeout=self.timeout)
+            exit_code = result.get("StatusCode", -1)
+            
+            # Get logs
+            logs = container.logs(stdout=True, stderr=True).decode('utf-8', errors='ignore')
+            
+            # Parse output and errors
+            stdout, stderr = self._parse_container_logs(logs)
+            
+            success = exit_code == 0
+            
+            logger.info(f"{'?' if success else '?'} Container execution completed with exit code {exit_code}")
```

Ruff (0.14.2)

Useless `if`-`else` condition

(RUF034)